drivers/bluetooth/hci/Kconfig

```diff
@@ -237,3 +237,31 @@ config BT_SILABS_HCI_BUFFER_MEMORY
 	help
 	  Select the size of allocated memory buffer for the Silicon Labs
 	  Bluetooth Library.
+
+config BT_NXP_CTLR
+	bool "NXP Bluetooth Controller"
+	select GPIO
+	depends on BT_H4
+	select CRC
+	help
+	  Enables support for NXP Bluetooth Controller.
+	  More inforamtion about NXP Bluetooth profuct could be found on
+	  https://www.nxp.com/products/wireless-connectivity/wi-fi-plus-bluetooth-plus-802-15-4:WIFI-BLUETOOTH
```

Comment:
Should this perhaps mention that this is an extension of the H:4 driver, rather than being a complete module in itself? It's kind of implicit through the `depends on BT_H4` but I think it'd be good to explicitly mention it, to not give the impression that this option provides a complete HCI driver.

In fact, considering that it's an extension of the H4 driver, would something like `BT_H4_NXP_SETUP` or `BT_H4_NXP_CTLR` perhaps make more sense as the option name?

Reply:
Updated